auth0/auth0-deploy-cli
# Document resources that need to be excluded on free tier tenants

### 🔧 Changes



Documented how to use Auth0 Deploy CLI with free tier tenants, after recent pricing changes. Paid features need to be excluded from management.

### 📚 References



- Resolves GitHub Issue #969
- Auth0 Community Post: https://community.auth0.com/t/error-exporting-auth0-configuration-there-must-be-a-verified-credit-card-on-file-to-perform-this-operation/147879


### 🔬 Testing

1. Use a free tier Auth0 tenant, without credit card information. 

2. Try exporting the tenant with a basic config such as:
```
{
    "AUTH0_DOMAIN": "<YOUR_TENANT_DOMAIN_HERE>",
    "AUTH0_CLIENT_ID": "<MACHINE_TO_MACHINE_CLIENT_ID_HERE>",
    "AUTH0_CLIENT_SECRET": "<MACHINE_TO_MACHINE_CLIENT_SECRET_HERE>",
}
```
You will encounter an error, due to trying to export a paid feature.
Error: Problem loading tenant data from Auth0 Forbidden: There must be a verified credit card on file to perform this operation

3. When exporting from a free tier tenant, you need to exclude paid features using a config such as:
```
{
    "AUTH0_DOMAIN": "<YOUR_TENANT_DOMAIN_HERE>",
    "AUTH0_CLIENT_ID": "<MACHINE_TO_MACHINE_CLIENT_ID_HERE>",
    "AUTH0_CLIENT_SECRET": "<MACHINE_TO_MACHINE_CLIENT_SECRET_HERE>",
    "AUTH0_EXCLUDED": ["logStreams", "customDomains"]
}
```
You should no longer encounter errors importing/exporting paid features from free tier tenants.


### 📝 Checklist
